### PR TITLE
Edition breakage lint for absolute paths starting with modules

### LINKED ISSUE #48722: Add epoch breakage lint for fully qualified paths that do not use `crate`, `self`, or `super`
We need to add a 2018 epoch lint for:

`use foo::bar::baz` and `::foo::bar::baz`  where `foo` is not an external crate name, `crate`, `self`, or `super`. It should suggest `use crate::foo::bar::baz;` or `::crate::foo::bar::baz` instead

### PR DESCRIPTION
**Pull request by Manishearth**
We plan to enable `extern_absolute_paths` in the 2018 edition. To allow for that, folks must transition their paths in a previous edition to the new one. This makes paths which import module contents via `use module::` or `::module::` obsolete, and we must edition-lint these.

https://internals.rust-lang.org/t/the-great-module-adventure-continues/6678/205?u=manishearth is the current plan for paths.

r? @nikomatsakis

Fixes #48722

### CODE REVIEW

File: src/librustc/lint/builtin.rs [hunk 1]
```diff
@@ -254,6 +254,13 @@ declare_lint! {
     "suggest using `dyn Trait` for trait objects"
 }
 
+declare_lint! {
+    pub FULLY_QUALIFIED_PATH_WITH_MODULE,
```
Comment (petrochenkov) [inline src/librustc/lint/builtin.rs]: "Fully qualified" is `<Type as Trait>::AssocItem`, `::a::b::c` is "absolute", that's why the feature is called `extern_absolute_paths`.

**Comment by Manishearth** [comment]
Added test. Ready for review.

I also discovered that `crate::` doesn't work for inline paths, you have to use `::crate::`. This is a bug.

**Comment by TimNN** [comment]
Ping from triage @nikomatsakis ! This PR needs your review.

**Review by nikomatsakis — APPROVED** [review]

**Comment by nagbot-rs** [comment]
@bors: retry

* crates.io was down

**Comment by aturon** [comment]
@bors: p=1

trying to land edition-related changes

**Pull request merged**